server/lib/activity-event.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (event, data, request, queryParams) {
+  data.op = 'activityEvent';
+  data.event = event;
+  data.userAgent = request.headers['user-agent'];
+
+  if (queryParams) {
+    queryParams.forEach(function (param) {
+      data[param] = request.query[param];
+    });
+  }
+
+  setImmediate(function () {
```

Comment:
From what I understand, `setImmediate` has the advantage of returning control to the event loop before it is invoked, which seems desirable in this case. `process.nextTick` is called before the event loop, isn't it?